### DISCUSSION TITLE
hypersaw toggle?

### DISCUSSION THREAD

**Discussion by roguestreak** [marcel-licence/esp32_basic_synth#101] [Q&A] [answered]
Hi again Marcel
I'm attempting to remap your Edirol controls across to suit my Novation SL mk2, and noticed there's only a toggle for mono/poly; is it possible to add a toggle for hypersaw, to change between the two waveform selectors and the hypersaw mode without a re-code?

Is there also an ARP on/off, or is that done via one of the the 8 arp sequences?  In the Arp video you seemed to press B9 to go back to "normal" mode, but in z_config that's set to null and I'm not finding a boolean in the code anywhere for this?

Thanks for all your work, Lee

**Comment by marcel-licence** [comment, marked as answer]
Hi Lee,

you can use Arp_Idle and Arp_Active to control its internal state. It seems I didn't complete integrate the control mechanism. It is too far in the past to remember xD

Regarding the hypersaw ... I have an update in my queue I would like to work on soon. At the moment there is no mechanism to toggle without compiling.

Hope this helps,
Marcel

**Reply by roguestreak** [reply]
Thanks Marcel, I'll attempt to get sound happening with this one and subscribe for any future updates for the hypersaw toggle!  Will see if I can delve deeper to figure out how to toggle the arp myself, but was checking if I was missing already existing buttons.
